### PR TITLE
Bug 1995612: Breadcrumbs link fix for block pool related pages

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
This PR is top of: https://github.com/openshift/console/pull/9722
Issue: https://issues.redhat.com/browse/RHSTOR-2090, https://issues.redhat.com/browse/RHSTOR-2073

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
save-13
save-12
save-11

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/block-pool-list-page.tsx [hunk 1]
```diff
@@ -146,7 +150,10 @@ const BlockPoolTableRow: React.FC<RowFunctionArgs> = ({ obj, index, key, style,
         className={blockPoolColumnInfo.name.classes}
         columnID={blockPoolColumnInfo.name.id}
       >
-        <ResourceLink kind={referenceFor(obj)} name={name} namespace={namespace} />
+        <ResourceIcon kind={referenceForModel(CephBlockPoolModel)} />
+        <Link to={to} className="co-resource-item__resource-name" data-test-operand-link={name}>
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/block-pool/block-pool-list-page.tsx:154]:
```suggestion
        <Link to={to} className="co-resource-item__resource-name" data-test={name}>
```
This is not an operand link. We are not in the operators page.

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/block-pool-list-page.tsx [hunk 2]
```diff
@@ -297,10 +304,12 @@ type BlockPoolListRowProps = {
   storageClasses: OcsStorageClassKind[];
   poolRawCapacity: PoolMetrics;
   poolCompressionSavings: PoolMetrics;
+  listPagePath: string;
 };
 
 type BlockPoolListProps = {
   data: StoragePoolKind[];
+  customData: any;
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/block-pool/block-pool-list-page.tsx:312]:
```suggestion
  customData: string;
```

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/block-pool-list-page.tsx (hunk 1 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/block-pool-list-page.tsx:154]: ack

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/block-pool-list-page.tsx (hunk 2 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/block-pool-list-page.tsx:312]: ack

**Comment by GowthamShanmugam** [comment]
@bipuladh Changes done

**Review by bipuladh — COMMENTED** [review]
/lgtm

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/create-block-pool.tsx [hunk 3]
```diff
@@ -41,8 +38,10 @@ const CreateBlockPool: React.FC<CreateBlockPoolProps> = ({ match }) => {
 
   const cephCluster: CephClusterKind = useDeepCompareMemoize(cephClusters[0], true);
 
-  const blockPoolPageUrl = url.replace('/~new', '');
-  const pathName = appName || 'BlockPools';
+  // OCS create pool page url ends with ~new, ODF create pool page ends with /create/~new
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/block-pool/create-block-pool.tsx:41]: This should be removed in 4.10

**Review by bipuladh — COMMENTED** [review]
/lgtm

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/create-block-pool.tsx (hunk 3 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/create-block-pool.tsx:41]: ack

**Pull request merged**